tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -1038,6 +1089,28 @@ class SPIRVEmitter : public ASTConsumer {
     return 0;
   }
 
+  /// Processes the given expr, casts the result into the given bool (vector)
+  /// type and returns the <result-id> of the casted value.
+  uint32_t castToBool(const Expr *expr, QualType toBoolType) {
+    // Converting to bool means comparing with value zero.
+
+    const QualType exprType = expr->getType();
+
+    // If the given expr is already a boolean or a vector of booleans, no
+    // casting is necessary.
+    if (exprType->isBooleanType() ||
```

Comment:
A function for this is better since it will be reused. See https://github.com/google/DirectXShaderCompiler/pull/94/files#diff-d00a9d7516b840ab223da9e8420a9844L28. Depending on which one goes in first, either you or me need to rebase. :)